### PR TITLE
Add openqa-cli documentation

### PR DESCRIPTION
**Pull request by kraih**
So far we don't have much client documentation, even though most openQA users will interact with it directly. This is my first attempt at collecting everything i could think of and turning it into a new guide. Still needs some cleanups and perhaps more diverse examples, but the main points are there.

Progress: https://progress.opensuse.org/issues/10316
Feature PR: https://github.com/os-autoinst/openQA/pull/2859

### CODE REVIEW

**Comment by kraih** [comment]
I did not touch the other guides yet (which also contain some small client sections). And i'm not sure where in the overall documentation it belongs. Placing it above Pitfalls is just a starting point.

File: docs/Client.asciidoc [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+
+By default the returned JSON is often compressed, for better performance, and
+can be hard to read if the response gets larger. But if you add the `--pretty`
+option (or `-p` for short), `openqa-cli` can reformat it for you into much more
+user friendy format, with newlines and indentation.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com --pretty jobs/overview
+----
```
Comment (kalikiana) [inline docs/Client.asciidoc:220]: An example would be nice, like above, to demonstrate the nicer looking JSON output.

File: docs/Client.asciidoc [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+:toclevels: 6
+:author: Sebastian Riedel
+
+There are two ways to interact with openQA as a user. The web UI and the REST
+API. In this guide we will focus on the latter. You've probably already seen a
+few examples of its use with `openqa-cli` earlier in the documentation.
+
+Here we will start again from the very beginning. To give you a more complete
+overview of its capabilities. To get started all you need is an openQA instance
+with a few jobs and `curl`. Just replace `openqa.example.com` in the examples
```
[the remark this answers is not in the document]
Comment (kraih) [reply docs/Client.asciidoc:14]: You also don't want random people cause load on O3 with test requests, or accidentally pasted destructive examples being run against O3 by people who have the credentials to do damage. 😉

File: docs/Client.asciidoc [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+----
+
+For ad-hoc use all `openqa-cli` subcommands use the `--apikey` and `--apisecret`
+options. Which will override whatever the config files may contain.
+
+[source,sh]
+----
+openqa-cli api --host http://openqa.example.com --apikey 1234567890ABCDEF \
+    --apisecret 1234567890ABCDEF isos/1
+----
```
Comment (Amrysliu) [inline docs/Client.asciidoc:93]: Seems in this example, the `apikey` and `apisecret` are not needed.  Changing an example with `-X post` should be better?

**Pull request merged**